gcsio/src/main/java/com/google/cloud/hadoop/gcsio/EventLoggingHttpRequestInitializer.java

```diff
@@ -0,0 +1,69 @@
+package com.google.cloud.hadoop.gcsio;
+
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
+
+import com.google.api.client.http.HttpRequest;
+import com.google.api.client.http.HttpRequestInitializer;
+import com.google.api.client.http.HttpResponse;
+import com.google.cloud.MonitoredResource;
+import com.google.cloud.logging.LogEntry;
+import com.google.cloud.logging.Logging;
+import com.google.cloud.logging.Payload.JsonPayload;
+import com.google.cloud.logging.Severity;
+import com.google.common.base.Stopwatch;
+import java.io.IOException;
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+public class EventLoggingHttpRequestInitializer implements HttpRequestInitializer {
+
+  private final Map<HttpRequest, Stopwatch> requestTracker = new ConcurrentHashMap<>();
```

Comment:
requests that timeout will never receive a response and hence they will be stuck in this map for ever. Do you think we should clean this map after some timeout?

Reply:
One option is to set a custom request-header and populate it with current timestamp. In the `log` method then read that header and diff it with the current timestamp.

Reply:
The requests with timeouts should still be captured here. I don't follow the use-case here.

Reply:
requests which timeout will never receive a HTTP response and hence the log method will not be called. Those request I guess will rot in this map.

Reply:
both client and server side timeouts should be captured by this interceptor. could you point me to docs/ why would you think timeout errors are skipped ?